src/software/backend/simulation/physics/physics_robot_test.cpp

```diff
@@ -0,0 +1,312 @@
+#include "software/backend/simulation/physics/physics_robot.h"
+
+#include <Box2D/Box2D.h>
+#include <gtest/gtest.h>
+#include <math.h>
+
+#include "shared/constants.h"
+#include "software/backend/simulation/physics/physics_ball.h"
+#include "software/test_util/test_util.h"
+#include "software/world/robot.h"
+
+TEST(PhysicsRobotTest, test_get_robot_with_timestamp)
+{
+    b2Vec2 gravity(0, 0);
+    auto world = std::make_shared<b2World>(gravity);
+
+    Robot robot_parameter(0, Point(0, 0), Vector(0, 0), Angle::zero(),
+                          AngularVelocity::zero(), Timestamp::fromSeconds(0));
+    PhysicsRobot physics_robot(world, robot_parameter);
+    auto robot = physics_robot.getRobotWithTimestamp(Timestamp::fromSeconds(3.3));
+
+    // TODO: CHECK ROBOT STATE
```

Comment:
TODO?

Reply:
Waiting on #1037 . Removed the `TODO` for now as it's non-blocking (for now)